scripts/nayduck.py

```diff
@@ -203,6 +207,24 @@ def main():
         'tests': list(tests)
     }
 
+    if args.run_locally:
+        for test in post["tests"]:
+            # TODO: use parses similar to nayduck, where is it?
+            fields = test.split(" ")
+
+            if fields[0] == "expensive":
+                print("running expensive test %s" % test)
+                # timeout field not implemented
+                if fields[1].startswith("--timeout="):
+                    del fields[1]
+
+                package = fields[1]
+                params = test.split('::')[-1].split(" ")
+                subprocess.check_call(
+                    ['cargo', 'test', '-p', package, '--tests'] + params)
+            else:
+                print("ignoring don't know how to run locally %s" % test)
```

Comment:
```
        index = 1
        while len(fields) > index and fields[index].starts_with('--'):
            index += 1

        ignored = fields[1:index]
        del fields[1:index]
        message = f'Running ‘{"".join(fields)}’'
        if ignored:
            message = f'{message}  (ignoring flags ‘{"".join(ignored)}’)
        print(message)

        if fields[0] == 'expensive':
            cmd = ('cargo', 'test', '-p', fields[1], '--test', fields[2],
                   '--features', 'expensive_tests', '--', '--exact',
                   *fields[3])
            cwd = None
        elif fields[0] in ('pytest', 'mocknet'):
            fields[0] = sys.executable
            cmd = fields
            cwd = 'pytest'
        else:
            print(f'Unrecognised test category ‘{fields[0]}’', file=sys.stderr)
            continue
        subprocess.check_call(cmd, cwd=cwd)
```

I can add better handling of the ignored flags (--timeout, --skip-build, --release and --remote) later.

Reply:
@mina86 I had to disable parts of the code:

1) I had to remove `--test`, to make "expensive" case work
```
                                                                                                                              
subprocess.CalledProcessError: Command '('cargo', 'test', '-p', 'near-client', '--test', 'near_client', '--features', 'expensive_tests', '--', '--exact', 'tests::catching_up::test_catchup_re
ceipts_sync_third_epoch')' returned non-zero exit status 101.
```

2) +"pytest" - we run tests inside 'pytest' folder.
Tests are inside `tests/sandbox/patch_state.py'`,
however we run 'pytest /sandbox/patch_state.py' command.


Traceback (most recent call last):
  File "./scripts/nayduck.py", line 269, in <module>
    main()
  File "./scripts/nayduck.py", line 263, in main
    run_locally(tests)
  File "./scripts/nayduck.py", line 209, in run_locally
    subprocess.check_call(cmd, cwd=cwd)
  File "/usr/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/usr/bin/python3', 'sandbox/patch_state.py', '--features', 'sandbox']' returned non-zero exit status 2.

Reply:
Looks like `--test` only works if the test is in `tests` not `src/tests`.

And yes, sorry, for pytest case it should be:

            fields[0] = sys.executable
            fields[1] = os.path.join('tests', fields[1])
            cmd = fields
            cwd = 'pytest'